### ISSUE TITLE
`refresh_token_validity` change crash with `aws_cognito_user_pool_client`

### ISSUE THREAD

**Issue by jayanderson** [terraform-providers/terraform-provider-aws#4848]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

```
→ terraform -v
Terraform v0.11.2
+ provider.aws v1.23.0
```

Though the same issue occurs with provider.aws `v1.7.1` where this resource was introduced and those in between.

### Affected Resource(s)

* aws_cognito_user_pool_client

### Terraform Configuration Files

```hcl
provider "aws" {
  version = "~> 1.23.0"
  region  = "us-east-1"
}

resource "aws_cognito_user_pool" "pool" {
  name = "test-pool-2018-06-15"
}

resource "aws_cognito_user_pool_client" "client" {
  name         = "client"
  user_pool_id = "${aws_cognito_user_pool.pool.id}"

  // Setting this causes the crash
  //refresh_token_validity = 100
}
```

### Debug Output

Console output: https://gist.github.com/jayanderson/bbcee352eb559e42b43bc4ed9bd7de1e

### Panic Output

console.log: https://gist.github.com/jayanderson/71e476c28bbdcef666db1a187a2d7817

### Steps to Reproduce

1. `terraform apply` the above terraform
1. Uncomment the `refresh_token_validity` change
1. `terraform apply` again -> crash

### Expected Behavior

It should have applied the change to `refresh_token_validity` successfully.

### Actual Behavior

It crashes. Along with the crash the full state of the `aws_cognito_user_pool_client` is wiped out. 😬

### Important Factoids

None that I know of. 

### References

None

**Comment by bflad** [comment]
Hi @jayanderson 👋 Thanks for reporting this and sorry you ran into the unexpected behavior here. 

This does seem like a bug in the resource that should have a fairly trivial fix.

Panic portion of the crash log:

```
2018-06-15T11:14:18.707-0700 [DEBUG] plugin.terraform-provider-aws_v1.23.0_x4: panic: interface conversion: interface {} is int, not int64
2018-06-15T11:14:18.707-0700 [DEBUG] plugin.terraform-provider-aws_v1.23.0_x4: 
2018-06-15T11:14:18.707-0700 [DEBUG] plugin.terraform-provider-aws_v1.23.0_x4: goroutine 127 [running]:
2018-06-15T11:14:18.707-0700 [DEBUG] plugin.terraform-provider-aws_v1.23.0_x4: github.com/terraform-providers/terraform-provider-aws/aws.resourceAwsCognitoUserPoolClientUpdate(0xc420abe150, 0x3510960, 0xc4205e4840, 0x24, 0x5875b20)
2018-06-15T11:14:18.707-0700 [DEBUG] plugin.terraform-provider-aws_v1.23.0_x4: 	/opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_cognito_user_pool_client.go:286 +0xdcb
2018-06-15T11:14:18.707-0700 [DEBUG] plugin.terraform-provider-aws_v1.23.0_x4: github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/schema.(*Resource).Apply(0xc420217c00, 0xc420953040, 0xc42094b660, 0x3510960, 0xc4205e4840, 0x1, 0xc4205e06f0, 0xc4205e06f0)
2018-06-15T11:14:18.707-0700 [DEBUG] plugin.terraform-provider-aws_v1.23.0_x4: 	/opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/schema/resource.go:233 +0x259
2018-06-15T11:14:18.707-0700 [DEBUG] plugin.terraform-provider-aws_v1.23.0_x4: github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/schema.(*Provider).Apply(0xc420676070, 0xc420952ff0, 0xc420953040, 0xc42094b660, 0x68621e8, 0x0, 0x18)
2018-06-15T11:14:18.707-0700 [DEBUG] plugin.terraform-provider-aws_v1.23.0_x4: 	/opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/schema/provider.go:283 +0xa4
2018-06-15T11:14:18.707-0700 [DEBUG] plugin.terraform-provider-aws_v1.23.0_x4: github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/plugin.(*ResourceProviderServer).Apply(0xc4209061a0, 0xc42094b220, 0xc4205a93b0, 0x0, 0x0)
2018-06-15T11:14:18.707-0700 [DEBUG] plugin.terraform-provider-aws_v1.23.0_x4: 	/opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/plugin/resource_provider.go:527 +0x57
2018-06-15T11:14:18.708-0700 [DEBUG] plugin.terraform-provider-aws_v1.23.0_x4: reflect.Value.call(0xc42051c660, 0xc42000d200, 0x13, 0x397fe6b, 0x4, 0xc420b81f20, 0x3, 0x3, 0x0, 0x0, ...)
2018-06-15T11:14:18.708-0700 [DEBUG] plugin.terraform-provider-aws_v1.23.0_x4: 	/usr/local/go/src/reflect/value.go:434 +0x905
2018-06-15T11:14:18.708-0700 [DEBUG] plugin.terraform-provider-aws_v1.23.0_x4: reflect.Value.Call(0xc42051c660, 0xc42000d200, 0x13, 0xc420731720, 0x3, 0x3, 0xc420165d40, 0x7a00000061, 0x0)
2018-06-15T11:14:18.708-0700 [DEBUG] plugin.terraform-provider-aws_v1.23.0_x4: 	/usr/local/go/src/reflect/value.go:302 +0xa4
2018-06-15T11:14:18.708-0700 [DEBUG] plugin.terraform-provider-aws_v1.23.0_x4: net/rpc.(*service).call(0xc42034a9c0, 0xc42092a3c0, 0xc420592a20, 0xc420a60100, 0xc420638940, 0x2ffa620, 0xc42094b220, 0x16, 0x2ffa660, 0xc4205a93b0, ...)
2018-06-15T11:14:18.708-0700 [DEBUG] plugin.terraform-provider-aws_v1.23.0_x4: 	/usr/local/go/src/net/rpc/server.go:381 +0x142
2018-06-15T11:14:18.708-0700 [DEBUG] plugin.terraform-provider-aws_v1.23.0_x4: created by net/rpc.(*Server).ServeCodec
2018-06-15T11:14:18.708-0700 [DEBUG] plugin.terraform-provider-aws_v1.23.0_x4: 	/usr/local/go/src/net/rpc/server.go:475 +0x36b
```

**Comment by bflad** [comment]
Bug fix pull request submitted: #4868

**Comment by bflad** [comment]
The fix for this has been merged into master and will release with version 1.24.0 of the AWS provider, likely middle of this week.

**Comment by jayanderson** [comment]
Awesome. Thanks. This was a very quick turnaround.

**Comment by bflad** [comment]
This has been released in [version 1.24.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1240-june-21-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
